cloud/amazon/ec2_vpc_dhcp_options.py

```diff
@@ -128,105 +157,204 @@
+    found = False
+    changed = False
+    new_options = collections.defaultdict(lambda: None)
 
   
     region, ec2_url, boto_params = get_aws_connection_info(module)
     connection = connect_to_aws(boto.vpc, region, **boto_params)
-  
-    inherit_existing = params['inherit_existing']
-  
-    existing_options = _get_associated_dhcp_options(params['vpc_id'], connection)
-    new_options = collections.defaultdict(lambda: None)
-  
-    new_options['domain-name-servers'] = _get_updated_option( params['dns_servers'],
-            existing_options.options.get('domain-name-servers'), inherit_existing)
-  
-    new_options['netbios-name-servers'] = _get_updated_option(params['netbios_name_servers'],
-            existing_options.options.get('netbios-name-servers'), inherit_existing)
-  
-  
-    new_options['ntp-servers'] = _get_updated_option(params['ntp_servers'],
-            existing_options.options.get('ntp-servers'), inherit_existing)
-  
-    # HACK: Why do I make the next two lists? The boto api returns a list if present, so
-    # I need this to properly compare so == works.
-  
-    # HACK: netbios-node-type is an int, but boto returns a string. So, asking for an int from Ansible
-    # for data validation, but still need to cast it to a string
-    new_options['netbios-node-type'] = _get_updated_option(
-            [str(params['netbios_node_type'])], existing_options.options.get('netbios-node-type'),
-            inherit_existing)
-  
-    new_options['domain-name'] = _get_updated_option(
-            [params['domain_name']], existing_options.options.get('domain-name'), inherit_existing)
-  
-    if existing_options and new_options == existing_options.options:
-        module.exit_json(changed=False)
-  
-    if new_options['netbios-node-type']:
-        new_options['netbios-node-type'] = new_options['netbios-node-type'][0]
-  
-    if new_options['domain-name']:
-        new_options['domain-name'] = new_options['domain-name'][0]
-  
-    if not module.check_mode:
-        dhcp_option = connection.create_dhcp_options(new_options['domain-name'],
-                new_options['domain-name-servers'], new_options['ntp-servers'],
-                new_options['netbios-name-servers'], new_options['netbios-node-type'])
+
+    # First check if we were given a dhcp_options_id
+    if not params['dhcp_options_id']:
+    # No, so create new_options from the parameters
+        if params['dns_servers'] != None:
+            new_options['domain-name-servers'] = params['dns_servers']
+        if params['netbios_name_servers'] != None:
+            new_options['netbios-name-servers'] = params['netbios_name_servers']
+        if params['ntp_servers'] != None:
+            new_options['ntp-servers'] = params['ntp_servers']
+        if params['domain_name'] != None:
+            # needs to be a list for comparison with boto objects later
+            new_options['domain-name'] = [ params['domain_name'] ]
+        if params['netbios_node_type'] != None:
```

Comment:
Because there's a possibility that the user chose to inherit, this is needed...
otherwise explicitly providing an empty list (intending to replace a populated one) would be ignored,and the list on the old dhcp options would be reapplied.

(e.g. the existing dhcp options contains an ntp server list but we want to remove that)